src/org/opendatakit/briefcase/util/ExportToCsv.java

```diff
@@ -134,6 +129,33 @@ public boolean doAction() {
 
     File[] instances = instancesDir.listFiles();
 
+    //Sorts the instances by the submission date. If no submission date, we assume it to be latest.
+    Arrays.sort(instances, new Comparator<File>() {
+      File submission1, submission2;
+      String submissionDate1String, submissionDate2String;
+      Date submissionDate1, submissionDate2;
+      public int compare(File f1, File f2) {
+        try {
+          submission1 = new File(f1, "submission.xml");

```

Comment:
We could potentially improve this by doing simple memoization of already parsed submission. That way we don't need to parse the same file again.